daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -336,28 +337,23 @@ private[speedy] case class PartialTransaction(
     */
   def insertCreate(
       submissionTime: Time.Timestamp,
-      templateId: Ref.Identifier,
-      arg: Value,
-      agreementText: String,
+      contract: CachedContract,
```

Comment:
Nice idea to refactor/simplify by passing in the cached contract (rather than its bits and pieces).